pkg/intermediate/aggregate.go

```diff
@@ -215,6 +215,71 @@ func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
 	return a.inactiveExpiryTimeout
 }
 
+// GetRecordStrings returns the flow records by FlowKey from flowKeyRecordMap.
+// Returns all the flow records if FlowKey is not provided.
+func (a *AggregationProcess) GetRecordStrings(flowKey *FlowKey) []string {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+
+	var records []string
```

Comment:
We can first have a size check of records, then initialize the string slice with fixed size and assign record by its index instead of using append. In that way we can save reduce memory allocations.

Reply:
The size of the records may depend on the flow key provided. Like if the user only provides a source address in the flow key, all the records with that source address will be return. Should I allocate the same size as the flowKeyRecordMap, which is an upper bound. Or should I iterate the map twice, once for the size and the other to store the string? Or is there any other way?

Reply:
I see your point. So the function does not actually return the flow records that have exact match with flow key right? If so, better to have some comment/description clarifying that. When we call the function, we will need to create a new flow key even we only want to search the match of sourceIPAddress?

Reply:
Yes to both questions. If we provide a complete flowkey, it would be an exact match, otherwise it is like a partial filter. I will add some descriptions. I am also going to add a hash check first as suggested by [Srika's comments above](https://github.com/vmware/go-ipfix/pull/261#discussion_r720428027).

Reply:
Updated description